### ISSUE TITLE
False positive for `arguments-differ` when *args and **kwargs are provided 

### ISSUE THREAD

**Issue by snejus** [PyCQA/pylint#3672]
There have been similar issues, like #1553 but they seem to be closed.

```python
class NoNullCharField(CharField):
    """CharField that respects the default `""` value and uses it instead of NULLs."""

    def get_db_prep_value(self, value, *args, **kwargs):  # signature-differs is emitted
        """Convert `None` to an empty string."""
        if all({self.blank, self.default == "", not self.null, value is None}):
            value = ""

        return super().get_db_prep_value(value, *args, **kwargs)
```
where 
```python
class CharField(Field):
    # does not define this method

class Field:
    ...
    def get_db_prep_value(self, value, connection, prepared=False):
        ...
```

### Current behavior
Message is shown.

### Expected behavior
Nothing is shown.

### pylint --version output
```bash
pylint 2.5.2
astroid 2.4.1
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0]
```
